mlflow/models/evaluation/default_evaluator.py

```diff
@@ -923,9 +924,10 @@ def evaluate(
                     f"verify that you set the `model_type` and `dataset` arguments correctly."
                 )
 
-            if model_type == "classifier":
-                return self._evaluate_classifier()
-            elif model_type == "regressor":
-                return self._evaluate_regressor()
-            else:
-                raise ValueError(f"Unsupported model type {model_type}")
+            with _AUTOLOGGING_METRICS_MANAGER.disable_log_post_training_metrics():
```

Comment:
@harupy Can we disable autologging entirely? Running SHAP might train kmeans models, which would still be logged to MLflow.

Reply:
Great idea, fixed!